lib/web_ui/test/engine/semantics/accessibility_test.dart

```diff
@@ -18,20 +21,29 @@ void main() {
 void testMain() {
   setUpAll(() async {
     await initializeEngine();
+    setLiveMessageDurationForTest(const Duration(milliseconds: 10));
   });
 
-  group('$AccessibilityAnnouncements', () {
-    void expectAnnouncementElements({required bool present}) {
-      expect(
-        domDocument.getElementById('ftl-announcement-polite'),
-        present ? isNotNull : isNull,
-      );
-      expect(
-        domDocument.getElementById('ftl-announcement-assertive'),
-        present ? isNotNull : isNull,
-      );
-    }
+  void expectAnnouncementElements({required bool present}) {
+    expect(
+      domDocument.getElementById('ftl-announcement-polite'),
+      present ? isNotNull : isNull,
+    );
+    expect(
+      domDocument.getElementById('ftl-announcement-assertive'),
+      present ? isNotNull : isNull,
+    );
+  }
+
+  tearDown(() async {
+    // Completely reset accessibility announcements for subsequent tests.
+    accessibilityAnnouncements.dispose();
+    await Future<void>.delayed(liveMessageDuration * 2);
+    initializeAccessibilityAnnouncements();
+    expectAnnouncementElements(present: true);
```

Comment:
I find it slighly weird to "initializeX" inside of a `tearDown` method.

Would it make sense that `initializeAccessibilityAnnouncements` and (maybe) `expectAnnouncementElements` are the implementation of a `setUp` method?

I suspect this being out of order makes the tests in this file dependent on the execution order? (what happens if you don't run the "Initialization and disposal" test?)

Reply:
Aggg too late :/

Reply:
Thanks for checking! Every test assumes that it's initialized at the start because that's what initializing the engine does on `setUpAll`. This is the same for the initialization and disposal test as well. So there should be no order dependence.

Reply:
@marcianx yes! thanks for the response!